### PR TITLE
Add cfgs (everywhere) in `liblibc` for NaCl targets.

### PR DESCRIPTION
**Pull request by DiamondLovesYou**
This does not allow Rust proper to target NaCl; this just adds support for NaCl
crosses to `liblibc` on crates.io.

### CODE REVIEW

**Comment by DiamondLovesYou** [comment]
This is preventing some of my projects from building.

r? @alexcrichton

**Comment by alexcrichton** [comment]
I'd like to start trying hard to cut down on the explosion of `#[cfg]` in liblibc as it really does seem to be getting out of hand. To that end, could you make a few modifications?

* The top-level `pub use` can all go back to `pub use foo::*` now that globs are stable.
* Each sub-block should avoid one-off `#[cfg]` (this PR adds a lot), if that happens then I think it should be separated to a whole new block.

**Comment by DiamondLovesYou** [comment]
@alexcrichton Done!

File: src/liblibc/lib.rs [hunk 1]
```diff
@@ -3152,7 +3016,6 @@ pub mod consts {
             pub const F_SETFL : c_int = 4;
 
             pub const SIGTRAP : c_int = 5;
-            pub const SIGPIPE: c_int = 13;
```
Comment (alexcrichton) [inline src/liblibc/lib.rs]: Shouldn't this module not be changing?

File: src/liblibc/lib.rs [hunk 2]
```diff
@@ -3202,11 +3065,21 @@ pub mod consts {
...
+            // Yeah...
+            #[cfg(target_os = "nacl")]
+            pub const PTHREAD_CREATE_JOINABLE: c_int = 1;
+            #[cfg(target_os = "nacl")]
+            pub const PTHREAD_CREATE_DETACHED: c_int = 0;
+
             #[cfg(target_os = "android")]
             pub const PTHREAD_STACK_MIN: size_t = 8192;
+            #[cfg(target_os = "nacl")]
+            pub const PTHREAD_STACK_MIN: size_t = 1024;
```
Comment (alexcrichton) [inline src/liblibc/lib.rs]: These changes are folded below, right?

File: src/liblibc/lib.rs [hunk 3]
```diff
@@ -3683,7 +3634,6 @@ pub mod consts {
             pub const F_SETFL : c_int = 4;
 
             pub const SIGTRAP : c_int = 5;
-            pub const SIGPIPE: c_int = 13;
```
Comment (alexcrichton) [inline src/liblibc/lib.rs]: Were these deletions accidental?

File: src/liblibc/lib.rs [hunk 4]
```diff
@@ -469,9 +322,15 @@ pub mod types {
                     #[cfg(target_os = "android")]
                     pub ai_canonname: *mut c_char,
 
+                    #[cfg(target_os = "nacl")]
```
Comment (alexcrichton) [inline src/liblibc/lib.rs]: Could this `#[cfg]` just move up to the previous cfg for android?

File: src/liblibc/lib.rs (hunk 1 above)
Comment (DiamondLovesYou) [inline src/liblibc/lib.rs]: SIGPIPE is exported in posix88, conflicting with this (and the others).

File: src/liblibc/lib.rs (hunk 3 above)
Comment (DiamondLovesYou) [inline src/liblibc/lib.rs]: Nope.

File: src/liblibc/lib.rs (hunk 4 above)
Comment (DiamondLovesYou) [inline src/liblibc/lib.rs]: Sure; I was just following the surrounding style.

File: src/liblibc/lib.rs (hunk 2 above)
Comment (DiamondLovesYou) [inline src/liblibc/lib.rs]: Yeah, looks like I just forget to fix this part.

**Comment by DiamondLovesYou** [comment]
@alexcrichton I've addressed your comments.

**Comment by alexcrichton** [comment]
@bors: r+ c70e4ce

Thanks!

**Comment by DiamondLovesYou** [comment]
@alexcrichton I've issued a possible fix (no local win machine available to test).

**Comment by alexcrichton** [comment]
Thanks! Could you squash the commits down together as well?

**Comment by DiamondLovesYou** [comment]
@alexcrichton Done!

**Comment by DiamondLovesYou** [comment]
@alexcrichton And again.

**Comment by Manishearth** [comment]
I suspect this to be the cause of a failure I'm getting on Android

http://buildbot.rust-lang.org/builders/auto-linux-64-x-android-t/builds/3681/steps/test/logs/stdio
http://buildbot.rust-lang.org/builders/auto-linux-64-x-android-t/builds/3685/steps/test/logs/stdio

error: undefined reference to 'pthread_atfork'

(Removed from the rollup until this gets fixed, sorry)

**Comment by Manishearth** [comment]
Intermittent. Retryable.

**Comment by Manishearth** [comment]
It's the `rm` intermittent again :/

```
rm: cannot remove ‘i686-pc-windows-gnu/stage2/test/stdtest-i686-pc-windows-gnu.0.o’: Device or resource busy
make: *** [clean-generic-H-i686-pc-windows-gnu] Error 123
```

**Comment by DiamondLovesYou** [comment]
@Manishearth Fixed merge conflict.

**Comment by Manishearth** [comment]
@bors: retry

(slave lost)

**Comment by DiamondLovesYou** [comment]
I didn't know it was possible to cause so much pain and suffering.

**Comment by Manishearth** [comment]
Not your fault, I blame the windows builders.

I'm looking up what the appropriate sacrifice  to appease the Windows gods  is right now. :crying_cat_face:

**Comment by huonw** [comment]
Something weird seems to be going on with https://github.com/rust-lang/rust/pull/22548, and this PR happened to be caught up. Closing temporarily.

**Comment by Manishearth** [comment]
(My fault, sorry about this. We'll get your PR through once the rollup passes.)

**Comment by Manishearth** [comment]
@bors: p=1

Let's get platform specific stuff out of the way first

**Comment by Manishearth** [comment]
@bors: retry

 - #22628

**Comment by Manishearth** [comment]
Yes!

**Comment by DiamondLovesYou** [comment]
\o/

**Pull request merged**